pallets/runtime/tests/src/storage.rs

```diff
@@ -949,3 +975,67 @@ macro_rules! assert_event_doesnt_exist {
         }));
     };
 }
+
+pub fn exec<C: Into<Call>>(origin: Origin, call: C) -> DispatchResult {
+    let origin: Result<RawOrigin<AccountId>, Origin> = origin.into();
+    let signed = match origin.unwrap() {
+        RawOrigin::Signed(acc) => {
+            let info = frame_system::Account::<TestStorage>::get(&acc);
+            Some((acc, signed_extra(info.nonce)))
+        }
+        _ => None,
+    };
+    Executive::apply_extrinsic(sign(CheckedExtrinsic {
+        signed,
+        function: call.into(),
+    }))
+    .unwrap()
+}
+
+/// Sign given `CheckedExtrinsic`.
+fn sign(xt: CheckedExtrinsic) -> UncheckedExtrinsic {
+    match xt.signed {
+        Some((signed, extra)) => {
+            let payload = (
+                xt.function,
+                extra.clone(),
+                VERSION.spec_version,
+                VERSION.transaction_version,
+                GENESIS_HASH,
+                GENESIS_HASH,
+            );
+            let key = AccountKeyring::from_account_id(&signed).unwrap();
+            let signature = payload
+                .using_encoded(|b| {
+                    if b.len() > 256 {
+                        key.sign(&sp_io::hashing::blake2_256(b))
+                    } else {
+                        key.sign(b)
+                    }
+                })
+                .into();
+            UncheckedExtrinsic {
+                signature: Some((Address::Id(signed), signature, extra)),
+                function: payload.0,
+            }
+        }
+        None => UncheckedExtrinsic {
+            signature: None,
+            function: xt.function,
+        },
+    }
+}
+
+/// Returns transaction extra.
+fn signed_extra(nonce: Index) -> SignedExtra {
+    (
+        frame_system::CheckSpecVersion::new(),
+        frame_system::CheckTxVersion::new(),
+        frame_system::CheckGenesis::new(),
+        frame_system::CheckEra::from(Era::mortal(256, 0)),
+        frame_system::CheckNonce::from(nonce),
+        polymesh_extensions::CheckWeight::new(),
+        pallet_transaction_payment::ChargeTransactionPayment::from(0),
+        pallet_permissions::StoreCallMetadata::new(),
+    )
+}
```

Comment:
Let's dedup this with other places that `SignedExtra` is being constructed in the code by taking in `nonce`, `tip`, `period`, and `current_block` as parameters.

Reply:
I'll fix the dupes in another PR because there are a decent number of them and would likely muddy up this PR.

Reply:
👍